> doctest failures appear to be related
> 
> ```
> FAILED bigframes/bigquery/_operations/ai.py::bigframes.bigquery._operations.ai.generate
> FAILED bigframes/bigquery/_operations/ai.py::bigframes.bigquery._operations.ai.generate_bool
> FAILED bigframes/bigquery/_operations/ai.py::bigframes.bigquery._operations.ai.generate_double
> FAILED bigframes/bigquery/_operations/ai.py::bigframes.bigquery._operations.ai.generate_int
> ```

Yeah those look like permission failures. I guess our doc runners don't have EUC permissions for AI functions.

--- a/bigframes/bigquery/_operations/ai.py
+++ b/bigframes/bigquery/_operations/ai.py
@@ -56,6 +56,7 @@ def generate(
 
         >>> import bigframes.pandas as bpd
         >>> import bigframes.bigquery as bbq
+        >>> bpd.options.display.progress_bar = "auto"
         >>> country = bpd.Series(["Japan", "Canada"])
         >>> bbq.ai.generate(("What's the capital city of ", country, " one word only"))
         0    {'result': 'Tokyo\\n', 'full_response': '{"cand...

--- a/bigframes/bigquery/_operations/ai.py
+++ b/bigframes/bigquery/_operations/ai.py
@@ -56,7 +56,6 @@ def generate(
 
         >>> import bigframes.pandas as bpd
         >>> import bigframes.bigquery as bbq
-        >>> bpd.options.display.progress_bar = "auto"
         >>> country = bpd.Series(["Japan", "Canada"])
         >>> bbq.ai.generate(("What's the capital city of ", country, " one word only"))
         0    {'result': 'Tokyo\\n', 'full_response': '{"cand...

--- a/notebooks/generative_ai/ai_functions.ipynb
+++ b/notebooks/generative_ai/ai_functions.ipynb
@@ -71,7 +71,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 1,
+   "execution_count": null,
    "id": "c9f924aa",
    "metadata": {},
    "outputs": [],
@@ -97,7 +97,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 2,
+   "execution_count": 3,
    "id": "471a47fe",
    "metadata": {},
    "outputs": [
@@ -105,7 +105,7 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "/usr/local/google/home/sycai/src/python-bigquery-dataframes/bigframes/core/global_session.py:113: DefaultLocationWarning: No explicit location is set, so using location US for the session.\n",
+      "/usr/local/google/home/sycai/src/python-bigquery-dataframes/bigframes/core/global_session.py:103: DefaultLocationWarning: No explicit location is set, so using location US for the session.\n",
       "  _global_session = bigframes.session.connect(\n"
      ]
     },
@@ -117,7 +117,7 @@
        "dtype: struct<result: string, full_response: extension<dbjson<JSONArrowType>>, status: string>[pyarrow]"
       ]
      },
-     "execution_count": 2,
+     "execution_count": 3,
      "metadata": {},
      "output_type": "execute_result"
     }

--- a/notebooks/generative_ai/ai_functions.ipynb
+++ b/notebooks/generative_ai/ai_functions.ipynb
@@ -71,7 +71,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 1,
    "id": "c9f924aa",
    "metadata": {},
    "outputs": [],
@@ -97,7 +97,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 3,
+   "execution_count": 2,
    "id": "471a47fe",
    "metadata": {},
    "outputs": [
@@ -105,7 +105,7 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "/usr/local/google/home/sycai/src/python-bigquery-dataframes/bigframes/core/global_session.py:103: DefaultLocationWarning: No explicit location is set, so using location US for the session.\n",
+      "/usr/local/google/home/sycai/src/python-bigquery-dataframes/bigframes/core/global_session.py:113: DefaultLocationWarning: No explicit location is set, so using location US for the session.\n",
       "  _global_session = bigframes.session.connect(\n"
      ]
     },
@@ -117,7 +117,7 @@
        "dtype: struct<result: string, full_response: extension<dbjson<JSONArrowType>>, status: string>[pyarrow]"
       ]
      },
-     "execution_count": 3,
+     "execution_count": 2,
      "metadata": {},
      "output_type": "execute_result"
     }

